File: py/medium/atoi-string-to-integer.py
def myAtoi(s: str) -> int:
    is_negative = False
    s = s.strip()

    if (s[0] == '-'):
        is_negative = True
        s = s[1:]
    elif(s[0] == '+'):
        s = s[1:]
    num = 0
    for i,v in enumerate(s):
        if (v >= '0' and v <= '9'):
            num = num * 10 + int(v)
        else:
            break

    if is_negative:
        num *= -1

    range = 2 ** 31
    if num < -range:
        return -range
    elif num > range -1:
        return range -1

    return num

print(myAtoi("-100de"))
print(myAtoi("9203"))
print(myAtoi("-67400 de"))
        

File: js/medium/atoi-string-to-integer.js
const myAtoi = (s) => {
  let isNegative = false;
  if (s[0] == "-") {
    isNegative = true;
    s = s.slice(1, s.length);
  } else if (s[0] == "-") {
    s = s.slice(1, s.length);
  }
  let num = 0;
  for (let char of s) {
    if (char >= "0" && char <= "9") {
      //   console.log("here", num, char);
      num = num * 10 + parseInt(char);
    } else {
      break;
    }
  }
  if (isNegative) num *= -1;
  return num;
};

console.log(myAtoi("-100de"));
console.log(myAtoi("9203"));
console.log(myAtoi("-67400 de"));
